### ISSUE TITLE
[BUG] srt-live-transmit not adhering to the timeout in certain cases

### ISSUE THREAD

**Issue by barhom** [Haivision/srt#1678]
We are testing srt-live-transmit to send a live stream from Sweden to Libya. Libya has a lossy connection and things run really good with SRT. Great job!

We have noticed a bug that only happens once every couple of days.

SRT is run as such on the "sender":

> srt-live-transmit -t 60 -tm 1 -v -s 5000 udp://239.100.0.126:1234 srt://{{SENDER-IP}}:10013?passphrase=XXXXXXXXXXXXXXXXX&mode=listener&maxbw=300000

We use timeout of 60 seconds just in case something goes wrong and expect srt-live-transmit to exit if that happens.

The issue is that once srt-live-transmit enters in this "unrecoverable" phase then it simply hangs with what you see in the log here below:

> ======= SRT STATS: sid=258962167
> PACKETS     SENT:        6199  RECEIVED:             0
> LOST PKT    SENT:        1251  RECEIVED:             0
> REXMIT      SENT:        1197  RECEIVED:             0
> DROP PKT    SENT:        4992  RECEIVED:             0
> FILTER EXTRA  TX:           0        RX:             0
> FILTER RX  SUPPL:           0  RX  LOSS:             0
> RATE     SENDING:     3.53076  RECEIVING:            0
> BELATED RECEIVED:           0  AVG TIME:             0
> REORDER DISTANCE:           0
> WINDOW      FLOW:        8139  CONGESTION:        8192  FLIGHT:          31
> LINK         RTT:   236.582ms  BANDWIDTH:   96.048Mb/s
> BUFFERLEFT:  SND:    11469000  RCV:           12286500
> 09:07:53.585726/SRT:RcvQ:w1*E:SRT.gg: @258962167:ATTACK/IPE: incoming ack seq -1 exceeds current 2047541505 by 99942141!
> 3025296 bytes lost, 66713680 bytes sent, 27965569452 bytes received
> 4717860 bytes lost, 66713680 bytes sent, 27967262016 bytes received
> 6516644 bytes lost, 66713680 bytes sent, 27969060800 bytes received
> 8236844 bytes lost, 66713680 bytes sent, 27970781000 bytes received
> 10035252 bytes lost, 66713680 bytes sent, 27972579408 bytes received

Looking at TCPDUMP on the port in order to see the activity I find the following,

```
tcpdump -i ens2.701 -n port 10013
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode

listening on ens2.701, link-type EN10MB (Ethernet), capture size 262144 bytes
10:32:10.919066 IP {CLIENT-IP}.56443 > {SENDER-IP}.10013: UDP, length 64
10:32:11.176455 IP  {CLIENT-IP}.56443 > {SENDER-IP}.10013: UDP, length 64
10:32:11.428514 IP  {CLIENT-IP}..33439 > {SENDER-IP}.10013: UDP, length 64

```

This shows that packets are being sent by the receiver side in order to re-establish the SRT tunnel but it is failing.

Currently the only solution is to kill the server-side and restart srt-live-transmit. The expected behaviour is that it should timeout according to the 60s timeout that I set.

**Comment by barhom** [comment]
Anyone ever encountered this?

**Comment by ethouris** [comment]
Please re-check with the latest version on master. I believe a fix was provided in #1673. There was also another fix that prevents this -1 ACK from being generated, but I have trouble finding it, but I believe there was a fix for it, too.

**Comment by barhom** [comment]
I'll test and reopen if it happens again.

Thanks
